service/resources/export.py

```diff
@@ -0,0 +1,122 @@
+"""Welcome export module"""
+import json
+import datetime
+import os
+import base64
+import pytz
+import falcon
+import jsend
+import sendgrid
+import sentry_sdk
+from ..modules.formio import Formio
+from ..transforms.export_submissions import ExportSubmissionsTransform
+from .hooks import validate_access
+
+@falcon.before(validate_access)
+class Export():
+    """Export class"""
+    def on_get(self, req, resp):
+        #pylint: disable=no-self-use,too-many-locals
+        """
+        on get request
+        return export message and response if successful
+        """
+        try:
+            timezone = pytz.timezone('America/Los_Angeles')
+
+            yesterday = datetime.datetime.now(timezone) - datetime.timedelta(days=1)
+            start_datetime_obj = datetime.datetime.combine(
+                yesterday, datetime.datetime.min.time())
+
+            # if start_date provided
+            if 'start_date' in req.params:
+                start_datetime_obj = datetime.datetime.strptime(
+                    req.params['start_date'], '%Y-%m-%d')
+
+            start_time_utc = timezone.localize(start_datetime_obj).astimezone(pytz.UTC)
+
+            # how many days we want included in report starting from start_date
+            report_days = int(req.params['days']) if 'days' in req.params else 1
+
+            end_datetime_obj = start_datetime_obj + datetime.timedelta(days=report_days)
+            end_time_utc = timezone.localize(end_datetime_obj).astimezone(pytz.UTC)
+
+            responses = Formio.get_formio_submission_by_query(
+                {
+                    'created__gte':start_time_utc.isoformat(),
+                    'created__lt':end_time_utc.isoformat(),
+                    'limit':2000*report_days
+                })
+
+            submissions_csv = ExportSubmissionsTransform().transform(responses)
+            msg = 'Export '+str(start_time_utc.isoformat())+' to '+str(end_time_utc.isoformat())
+
+            send_email = bool(req.params['send_email']) if 'send_email' in req.params else False
+            if send_email:
+                self.email(
+                    "Export "+str(start_datetime_obj.date()),
+                    content=msg,
+                    file_name="export-"+str(start_datetime_obj.date())+".csv",
+                    file_content=submissions_csv)
+
+            resp.body = json.dumps(jsend.success({'message': msg, 'responses':len(responses)}))
+            resp.status = falcon.HTTP_200
+
+        #pylint: disable=broad-except
+        except Exception as exception:
+            resp.status = falcon.HTTP_500
+            msg_error = repr(exception)
+            sentry_sdk.capture_message(msg_error)
+            resp.body = json.dumps(jsend.error(msg_error))
+
+    def email(self, subject, content="Hi", file_name=None, file_content=None):
```

Comment:
It seems like we have to send over 2 API keys (Apigee + Sendgrid) to accomplish the same thing. And I don't know what the state of email microservice is. Currently, it looks like it is showing a Fail build and only deployed on Staging.  

With that said, we should be able to iterate on change the `send_email` method to call our email microservice instead of SendGrid directly relatively painlessly when we want to switch over.